### ISSUE TITLE
OffCanvas in Loopgrid

### ISSUE THREAD

**Issue by gitalexa** [elementor/elementor#29049]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

using a offcanvas widget in a template for loopitem shows alwas the vales from the last queried post

### Steps to reproduce

put loop-widget to page
create loop-item for this loop.
in loopitem put button and offcanvas
set button to open offcanvas
in offcanvas set heading with dynamic content post-title

### Expected behavior

every off canvas in every loop item shodl show the title of the undelying post

### Isolating the problem

- [X] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [X] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt

== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MySQL Community Server - GPL v8.0.33
	PHP Version: 7.4.33
	PHP Memory Limit: 512M
	PHP Max Input Vars: 4000
	PHP Max Post Size: 64M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.5.5
	Site URL: https://www.hostingplusx.de
	Home URL: https://www.hostingplusx.de
	WP Multisite: No
	Max Upload Size: 64 MB
	Memory limit: 40M
	Max Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: de_DE_formal
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 1.0.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 3.0.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: de-DE
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36 Edg/130.0.0.0

== Active Plugins ==
	Akismet Anti-spam: Spam Protection
		Version: 5.3.3
		Author: Automattic - Anti-spam Team

	Elementor
		Version: 3.24.7
		Author: Elementor.com

	Elementor Forms Optigem Action
		Version: 1.0.0
		Author: Alexander Probst

	Elementor Pro
		Version: 3.24.4
		Author: Elementor.com

	eRecht24 legal texts for WordPress
		Version: 3.3.11
		Author: eRecht24, Weslink GmbH

	Head, Footer and Post Injections
		Version: 3.2.8
		Author: Stefano Lissa

	Header Footer Code Manager
		Version: 1.1.39
		Author: DraftPress

	Imagify
		Version: 2.2.2
		Author: Imagify – Optimize Images & Convert WebP & Avif

	JetElements For Elementor
		Version: 2.7.0
		Author: Crocoblock

	JetEngine
		Version: 3.5.7
		Author: Crocoblock

	JetFormBuilder
		Version: 3.4.2
		Author: Crocoblock

	JetPopup
		Version: 2.0.7
		Author: Crocoblock

	JetTricks
		Version: 1.4.10
		Author: Crocoblock

	ManageWP - Worker
		Version: 4.9.20
		Author: GoDaddy

	Math Captcha for Elementor Forms
		Version: 1.1.0
		Author: Albano Toska

	NinjaFirewall (WP Edition)
		Version: 4.6.1
		Author: The Ninja Technologies Network

	Rank Math SEO
		Version: 1.0.230
		Author: Rank Math

	Real Cookie Banner
		Version: 4.8.4
		Author: devowl.io

	Temporary Login Without Password
		Version: 1.8.3
		Author: StoreApps

	WooCommerce
		Version: 9.3.3
		Author: Automattic

	WP24 Domain Check
		Version: 1.10.12
		Author: WP24

	WP Mermaid
		Version: 1.0.2
		Author: Terry Lin

	WP Rocket
		Version: 3.17.1
		Author: WP Media

	WPvivid Backup Plugin
		Version: 0.9.107
		Author: WPvivid Team

	WPvivid Imgoptim Free
		Version: 0.9.23
		Author: WPvivid Team

	WPvivid Plugins Pro
		Version: 2.2.28
		Author: wpvivid.com

	WPvivid Staging
		Version: 2.0.19
		Author: wpvivid.com


== Must-Use Plugins ==
	ManageWP - Worker Loader
		Version: 1.0.0
		Author: GoDaddy

	NinjaFirewall WP (mu-plugin)
		Version: 1.0
		Author: The Ninja Technologies Network


== Verwendung der Elemente ==
	
	container : 0
		accordion : 1
		button : 1
		container : 9
		heading : 7
		html : 2
		icon : 2
		icon-list : 1
		jet-listing-dynamic-field : 2
		jet-listing-grid : 1
		jet-scroll-navigation : 1
		text-editor : 2
	page : 5
		container : 4
		heading : 1
		image : 1
		jet-engine-booking-form : 3
		jet-engine-profile-menu : 3
		jet-listing-grid : 1
	section : 11
		button : 6
		divider : 2
		google_maps : 1
		heading : 28
		icon : 1
		icon-list : 7
		image : 6
		image-box : 3
		progress : 4
		text-editor : 11
	wp-page : 19
		accordion : 6
		button : 3
		container : 29
		divider : 13
		heading : 44
		icon-box : 5
		icon-list : 1
		image : 3
		image-box : 7
		jet-engine-booking-form : 1
		jet-hotspots : 1
		jet-listing-grid : 2
		jet-slider : 1
		loop-grid : 1
		menu-anchor : 1
		shortcode : 5
		text-editor : 26
	loop-item : 1
		button : 1
		container : 4
		heading : 5
		off-canvas : 1
	jet-popup : 1
		button : 1
		container : 4
		heading : 2
	jet-listing-items : 0
		button : 1
		container : 4
		heading : 2
		jet-listing-dynamic-field : 2
		off-canvas : 1


== Einstellungen ==
	
	cpt_support: post, page, jet-popup
	allow_tracking: yes


== Eigenschaften ==
	Custom Fonts: 0
	Custom Icons: 0

== Integration ==
	
	recaptcha_v3: Aktiv
	woocommerce: Aktiv


== Elementor-Experimente ==
	Verbessertes Laden von CSS: Standardmäßig inaktiv
	Inline Font Icons: Standardmäßig inaktiv
	Zusätzliche individuelle Breakpoints: Aktiv
	Flexbox-Container: Aktiv
	Raster-Container: Aktiv
	Upgrade Swiper Library: Aktiv
	Nested Elements Performance: Aktiv
	Optimized Control Loading: Standardmäßig inaktiv
	Optimized Markup: Standardmäßig inaktiv
	Conditionally load Swiper CSS files: Standardmäßig inaktiv
	Plugin Onboarding: Aktiv
	Default to New Theme Builder: Aktiv
	Hello Theme Header &amp; Footer: Standardmäßig inaktiv
	Elementor Home Screen: Aktiv
	Build with AI: Aktiv
	Landingpages: Aktiv
	Launchpad Checklist: Standardmäßig inaktiv
	Verschachtelte Elemente: Aktiv
	Obere Editor-Leiste: Aktiv
	Pages Panel: Standardmäßig inaktiv
	Element Caching: Aktiv
	Link In Bio: Aktiv
	Floating Buttons: Aktiv
	Elementor Editor Events: Standardmäßig inaktiv
	Atomic Widgets: Standardmäßig inaktiv
	Bedingungen anzeigen: Aktiv
	Form Submissions: Aktiv
	Menü: Aktiv


== Protokoll ==
	
Log: showing 20 of 372024-09-13 09:43:40 [info] elementor-pro::elementor_pro_updater Started 
2024-09-13 09:43:40 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-09-13 09:43:40 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-09-13 09:43:40 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.23.1',
  'to' => '3.24.0',
)]
2024-09-13 14:27:39 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.1',
  'to' => '3.24.2',
)]
2024-09-13 14:27:40 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.24.0',
  'to' => '3.24.1',
)]
2024-09-13 14:27:40 [info] elementor::elementor_updater Started 
2024-09-13 14:27:40 [info] Elementor/Upgrades - _on_each_version Start  
2024-09-13 14:27:41 [info] Elementor/Upgrades - _on_each_version Finished 
2024-09-13 14:27:41 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.1',
  'to' => '3.24.2',
)]
2024-09-13 14:27:43 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.1',
  'to' => '3.24.2',
)]
2024-09-13 14:27:43 [info] elementor-pro::elementor_pro_updater Started 
2024-09-13 14:27:43 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-09-13 14:27:43 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-09-13 14:27:44 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.24.0',
  'to' => '3.24.1',
)]
2024-09-13 14:27:45 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.24.0',
  'to' => '3.24.1',
)]
2024-10-25 14:36:09 [info] elementor::elementor_updater Started 
2024-10-25 14:36:09 [info] Elementor/Upgrades - _on_each_version Start  
2024-10-25 14:36:09 [info] Elementor/Upgrades - _on_each_version Finished 
2024-10-25 14:36:09 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.24.2',
  'to' => '3.24.7',
)]

PHP: showing 3 of 3PHP: 2024-10-24 09:28:58 [notice X 2][../wp-content/plugins/elementor/includes/base/controls-stack.php::1487] Undefined index: _flex_size_tablet [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-10-25 14:38:57 [notice X 2][../wp-content/plugins/elementor/modules/history/revisions-manager.php::363] Undefined index: editor_post_id [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-10-25 14:46:13 [notice X 1][../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined index: condition_type [array (
  'trace' => '
#0: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: ../wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]

JS: showing 3 of 3JS: 2024-10-25 14:38:55 [error X 1][../wp-content/plugins/elementor/assets/js/packages/editor-documents/editor-documents.min.js?ver=3.24.7:1:5030] null is not an object (evaluating &#039;d().getCurrent().editor&#039;) 
JS: 2024-10-25 14:41:12 [error X 1][../wp-content/plugins/elementor/assets/js/packages/editor-documents/editor-documents.min.js?ver=3.24.7:1:5030] Cannot read properties of null (reading &#039;editor&#039;) 
JS: 2024-10-25 14:42:42 [error X 9][../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.24.7:3:649468] Cannot read properties of null (reading &#039;data&#039;) 



== Elementor - Compatibility Tag ==
	
	Elementor Forms Optigem Action: Kompatibilität nicht angegeben
	Elementor Pro: Kompatibilität nicht angegeben
	JetElements For Elementor: Kompatibilität nicht angegeben
	Math Captcha for Elementor Forms: Kompatibilität nicht angegeben

== Elementor Pro - Compatibility Tag ==

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by hxinen** [comment]
I too have the same issue.

**Comment by nicholaszein** [comment]
Hello there!

Thank you for your submission! 🙏 

### 🔗 🚀 There is already a similar request submitted by someone here #7106

Feel free to continue the conversation and cast your **⬆️ Upvote** there! 👉

Kind regards

**Comment by nicholaszein** [comment]
Hi @gitalexa and @hxinen!

This was more of a Feature Request, not a bug, as up until version 3.26.0 of Elementor Pro, this feature was not implemented in Elementor yet. 

But we have great news! 🙌 

### 📢 We're happy to announce that this issue was resolved with the release of Elementor Pro v3.26.0! 🥳 

You can now use the Off-Canvas widget inside Loop Grids to load additional content that is triggered Dynamically and Independently from each loop item.

You can learn more about how to use the Off-Canvas widget in this link: https://elementor.com/help/off-canvas-widget/

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
